### DISCUSSION TITLE
Waybar + Eww on multiple screens

### DISCUSSION THREAD

**Discussion by radu-corbu** [elkowar/eww#1046] [Q&A]
Hi,

I want to open an Eww widget when I click on a module from Waybar and have it open on the screen I clicked on (dual screen setup). 

More specifically, I want to open an Action Center widget when I press the microphone/audio/bluetooth/wifi/brightness/battery modules on my Waybar. I'm using the action center widget from [this repo](https://github.com/MrVivekRajan/Hypr-Dots/tree/main/Stylish/.config/eww).

My problem is I sometimes use a second screen for my laptop, and Eww is set to only open on one of the screens. What I want is for Eww to open on the currently active screen. I've looked through the docs but haven't found anything specific. 

There was one section in the docs that talked about passing `:monitor <primary>` to the widget but that doesn't seem to work, and I don't think it would've done what I want anyway. I've also looked this issue up online and found [this discussion](https://github.com/elkowar/eww/discussions/592), but I don't understand how to set up the widget to do what I want.

I know one solution _can_ be to create 2 entries for the action center widget, one with `:monitor 0` and another with `:monitor 1`. But, to my knowledge, when I click on a Waybar module, it doesn't have any information about what screen it's being displayed on, so I can't tell which of the 2 Eww widgets I'm supposed to open.

Does anyone have any advice?

Thanks!

**Comment by MatthieuBarthel** [comment]
Hi,

You can specify the monitor using the  `--screen` param when calling `eww open`, you don't need to create two widgets.

You should be able to find your active monitor using your window manager cli, on hyprland you can do something like that:

```sh
eww open myEwwWindow --screen $(hyprctl activewindow -j | jq '.monitor')
```

Hope it helps :)

**Reply by radu-corbu** [reply]
Hey,

Unfortunately, I've moved away from Hyprland, so I can't try out the proposed solution anymore. Hopefully someone else is in my shoes and will be helped by the code snippet!

Thank you.

**Reply by hashdefault** [reply]
> Hi,
> 
> You can specify the monitor using the `--screen` param when calling `eww open`, you don't need to create two widgets.
> 
> You should be able to find your active monitor using your window manager cli, on hyprland you can do something like that:
> 
> ```shell
> eww open myEwwWindow --screen $(hyprctl activewindow -j | jq '.monitor')
> ```
> 
> Hope it helps :)

it works perfectly, thanks.

**Comment by hashdefault** [comment]
Found another command that open even if there's no active window on the screen.
```
eww open myEwwWindow --screen $(hyprctl monitors -j | jq -r '.[] | select(.focused == true) | .name')
```
